File: app/src/main/java/tpcreative/co/qrscanner/common/SaveSingleton.java
package tpcreative.co.qrscanner.common;

public class SaveSingleton {
    private static SaveSingleton instance ;
    private SingletonSaveListener listener;

    public static SaveSingleton getInstance(){
        if (instance==null){
            synchronized (SaveSingleton.class){
                if (instance==null){
                    instance = new SaveSingleton();
                }
            }
        }
        return instance;
    }

    public void setListener(SingletonSaveListener listener){
        this.listener = listener;
    }

    public void reLoadData(){
        if (listener!=null){
            listener.reLoadData();
        }
    }

    public interface SingletonSaveListener{
        void reLoadData();
    }


}

File: app/src/main/java/tpcreative/co/qrscanner/common/SaveSingleton.kt
package tpcreative.co.qrscanner.common
class SaveSingleton {
    private var listener: SingletonSaveListener? = null
    fun setListener(listener: SingletonSaveListener?) {
        this.listener = listener
    }

    fun reloadData() {
        if (listener != null) {
            listener?.reloadData()
        }
    }

    interface SingletonSaveListener {
        fun reloadData()
    }

    companion object {
        private var instance: SaveSingleton? = null
        fun getInstance(): SaveSingleton? {
            if (instance == null) {
                synchronized(SaveSingleton::class.java) {
                    if (instance == null) {
                        instance = SaveSingleton()
                    }
                }
            }
            return instance
        }
    }
}